SDK_dotnet/Clients/GluwaClient.cs

```diff
@@ -374,24 +418,116 @@ private string getContractAddress(ECurrency currency)
 
             return encodedData;
         }
 
+        private string getBtcAddressSignature(string privateKey)
+        {
+            BitcoinSecret secret = new BitcoinSecret(privateKey, mNetwork);
+            string timestamp = new DateTimeOffset(DateTime.UtcNow).ToUnixTimeSeconds().ToString();
+            string signature = secret.PrivateKey.SignMessage(timestamp);
+
+            string signatureToEncode = $"{timestamp}.{signature}";
+            byte[] signatureByte = Encoding.UTF8.GetBytes(signatureToEncode);
+            string encodedData = Convert.ToBase64String(signatureByte);
+
+            return encodedData;
+        }
+
+        private string getGluwacoinTransactionSignature(ECurrency currency, string amount, string fee, string nonce, string address, string target, string privateKey)
+        {
+            BigInteger convertAmount = GluwacoinConverter.ConvertToGluwacoinBigInteger(amount);
+            BigInteger convertFee = GluwacoinConverter.ConvertToGluwacoinBigInteger(fee.ToString());
+
+            ABIEncode abiEncode = new ABIEncode();
+            byte[] messageHash = abiEncode.GetSha3ABIEncodedPacked(
+                new ABIValue("address", getContractAddress(currency)),
+                new ABIValue("address", address),
+                new ABIValue("address", target),
+                new ABIValue("uint256", convertAmount),
+                new ABIValue("uint256", convertFee),
+                new ABIValue("uint256", int.Parse(nonce))
+                );
+
+            EthereumMessageSigner signer = new EthereumMessageSigner();
+            string signature = signer.Sign(messageHash, privateKey);
+
+            return signature;
+        }
+
+        private async Task<string> getBtcTransactionSignatureAsync(ECurrency currency, string address, string amount, string fee, string target, string privateKey)
+        {
+            Result<BalanceResponse, ErrorResponse> getUnspentOutput = await GetBalanceAsync(currency, address, true);
+            List<UnspentOutput> unspentOutputs = getUnspentOutput.Data.UnspentOutputs.OrderByDescending(u => u.Amount).ToList();
+
+            Money amountValue = Money.Parse(amount);
+            Money feeValue = Money.Parse(fee);
+            Money totalAmountAndFeeValue = amountValue + feeValue;
+            BigInteger totalAmountAndFee = new BigInteger(totalAmountAndFeeValue.Satoshi);
+
+            BitcoinAddress sourceAddress = BitcoinAddress.Create(address, mNetwork);
+            BitcoinAddress targetAddress = BitcoinAddress.Create(target, mNetwork);
+            BitcoinSecret secret = new BitcoinSecret(privateKey, mNetwork);
+
+            List<UnspentOutput> usingUnspentOutputs = new List<UnspentOutput>();
+            BigInteger unspentOutputTotalAmount = BigInteger.Zero;
+            for (int i = 0; i < unspentOutputs.Count; i++)
+            {
+                if (unspentOutputTotalAmount >= totalAmountAndFee || i >= MAX_UNSPENTOUTPUTS_COUNT)
```

Comment:
Do we really need to check `MAX_UNSPENTOUTPUTS_COUNT` twice, here and a few lines below? How about just throw an exception from inside the for loop if the list goes over the max count and the amount is not yet reached?